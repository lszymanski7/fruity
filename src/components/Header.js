import React, { useState } from 'react'
import ToggleSwitch from './ToggleSwitch'
import { logos } from '../data/constants'

const Header = () => {
    // Application name
    const name = 'FRUITY'

    // Toggle switch states for dark and light mode
    const [isChecked, setIsChecked] = useState(false)

    return (
        <header className="flexbox-column header">
            <div className="flexbox-row header__widgets">
                <a href="https://github.com/lszymanski7/fruity-app">
                    <img
                        alt="GitHub Logo"
                        className="header__github-logo"
                        src={logos.github}
                    />
                </a>
                <ToggleSwitch
                    handleOnChange={() => setIsChecked(!isChecked)}
                    icons={['moon-icon', 'sun-icon']}
                    id="themeSwitcher"
                    isChecked={isChecked}
                />
            </div>
            <div className="flexbox-row header__brand">
                <img
                    alt="Fruity Logo"
                    className="header__app-logo"
                    src={logos.fruity.size_256x256}
                />
                <h1 className="header__app-name">{name}</h1>
            </div>
        </header>
    )
}

export { Header as default }
