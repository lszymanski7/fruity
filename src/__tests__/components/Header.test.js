import React from 'react'
import { render, screen } from '@testing-library/react'
import Header from '../../components/Header'

describe('Header', () => {
	it('should be rendered correctly.', () => {
		const component = render(<Header />)
		expect(component).toMatchSnapshot()
	})

	it('should have the correct title.', () => {
		render(<Header />)
		const title = 'Fruity'
		const h1 = screen.getByRole('heading', { level: 1 })
		expect(h1).toHaveTextContent(title)
	})

	it('should have the correct subtitle.', () => {
		render(<Header />)
		const subtitle = `Are you a fruit lover? Have you ever been unable to decide what kind of fruit you would like to eat? Don't worry about that anymore! Just click the button below!`
		const h2 = screen.getByRole('heading', { level: 2 })
		expect(h2).toHaveTextContent(subtitle)
	})

	it('should have a logo.', () => {
		render(<Header />)
		const logo = screen.getByRole('img')
		expect(logo).toHaveAttribute('alt', 'Fruity Logo')
		expect(logo).toBeInTheDocument()
	})
})
