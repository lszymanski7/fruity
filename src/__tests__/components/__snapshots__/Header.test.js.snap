// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="header-container"
          >
            <img
              alt="Fruity Logo"
              class="header__logo"
              src="logo512-outlined.png"
            />
            <h1
              class="header__title"
            >
              Fruity
            </h1>
          </div>
          <div>
            <h2
              class="header__subtitle"
            >
              Are you a fruit lover? Have you ever been unable to decide what kind of fruit you would like to eat? Don't worry about that anymore! Just click the button below!
            </h2>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header-container"
        >
          <img
            alt="Fruity Logo"
            class="header__logo"
            src="logo512-outlined.png"
          />
          <h1
            class="header__title"
          >
            Fruity
          </h1>
        </div>
        <div>
          <h2
            class="header__subtitle"
          >
            Are you a fruit lover? Have you ever been unable to decide what kind of fruit you would like to eat? Don't worry about that anymore! Just click the button below!
          </h2>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
